malt/converters/slices.py

```diff
@@ -31,17 +31,22 @@ class SliceTransformer(converter.Base):
   """
 
   def _process_single_assignment(self, target, value):
+    # (dime10) This function has been modified to support slices.
     if not isinstance(target, gast.Subscript):
       return None
     s = target.slice
-    if isinstance(s, (gast.Tuple, gast.Slice)):
+    if isinstance(s, (gast.Tuple)):
+      # multi-dimensional indices are not supported
       return None
 
     template = """
       target = ag__.set_item(target, key, item)
     """
+
+    if isinstance(s, (gast.Slice)):
+      template = template.replace("key", "slice(lower, upper, step)")
     return templates.replace(
-        template, target=target.value, key=target.slice, item=value)
+      template, target=target.value, key=target.slice, lower=s.lower, upper=s.upper, step=s.step, item=value)
```

Comment:
Looks much better, thanks @tzunghanjuang! 

Just confirming, but `templates.replace` won't fail if lower/upper/step don't exist right?

Reply:
`templates.replace` will access the visit methods from `ReplaceTransformer`.  key and lower/upper/step will be stored in the ``replacements`` dictionary.
https://github.com/PennyLaneAI/diastatic-malt/blob/8a09938e9054e352b749d20522bc9aeb173307fd/malt/pyct/templates.py#L266-L271

In the visit methods, we have a guard like this to check if a node matches any replacement.. So I think It is safe to pass unused replacement.
https://github.com/PennyLaneAI/diastatic-malt/blob/8a09938e9054e352b749d20522bc9aeb173307fd/malt/pyct/templates.py#L194-L196

Reply:
Wonderful 💯